hazelcast/proxy/vector_collection.py

```diff
@@ -0,0 +1,282 @@
+        self._reference_id_generator = context.lock_reference_id_generator
+
+    def blocking(self) -> "BlockingVectorCollection":
+        return BlockingVectorCollection(self)
+
+    def get(self, key: Any) -> Future[Optional[Document]]:
+        check_not_none(key, "key can't be None")
+        return self._get_internal(key)
+
+    def set(self, key: Any, document: Document) -> Future[None]:
+        check_not_none(key, "key can't be None")
+        check_not_none(document, "document can't be None")
+        check_not_none(document.value, "document value can't be None")
+        return self._set_internal(key, document)
+
+    def put(self, key: Any, document: Document) -> Future[Optional[Document]]:
+        check_not_none(key, "key can't be None")
+        check_not_none(document, "document can't be None")
+        check_not_none(document.value, "document value can't be None")
+        return self._put_internal(key, document)
+
+    def put_all(self, map: Dict[Any, Document]) -> Future[None]:
+        check_not_none(map, "map can't be None")
+        if not map:
+            return ImmediateFuture(None)
+        partition_service = self._context.partition_service
+        partition_map: Dict[str, List[Tuple[Data, Document]]] = {}
+
+        for key, doc in map.items():
+            check_not_none(key, "key can't be None")
+            check_not_none(doc, "value can't be None")
+            doc = copy.copy(doc)
+            doc.value = self._to_data(doc.value)
+            try:
+                entry = (self._to_data(key), doc)
+            except SchemaNotReplicatedError as e:
+                return self._send_schema_and_retry(e, self.put_all, map)
+
+            partition_id = partition_service.get_partition_id(entry[0])
+            try:
+                partition_map[partition_id].append(entry)
+            except KeyError:
+                partition_map[partition_id] = [entry]
+
+        futures = []
+        for partition_id, entry_list in partition_map.items():
+            request = vector_collection_put_all_codec.encode_request(self.name, entry_list)
+            future = self._invoke_on_partition(request, partition_id)
+            futures.append(future)
+
+        return combine_futures(futures)
+
+    def put_if_absent(self, key: Any, document: Document) -> Future[Optional[Document]]:
+        check_not_none(key, "key can't be None")
+        check_not_none(document, "document can't be None")
+        check_not_none(document.value, "document value can't be None")
+        return self._put_if_absent_internal(key, document)
+
+    def search_near_vector(
+        self,
+        vector: Vector,
+        *,
+        include_value: bool = False,
+        include_vectors: bool = False,
+        limit: Optional[int] = None
+    ) -> Future[List[SearchResult]]:
+        check_not_none(vector, "vector can't be None")
+        if limit is None:
+            limit = 2147483629
```

Comment:
this is 2 ** 31 - 19

I wonder what's special about this number. consider introducing a constant and explain it via constant's name or docstring

Reply:
I set the default to `-1` like in the Java client.